### PR TITLE
Phase 4b/4c: shaka-lab-gateway-client for Windows and macOS

### PR DESCRIPTION
**Pull request by joeyparrish**
The gateway client package configures devices in the lab to allow centrally-managed Active Directory users to log in.

### CODE REVIEW

**Comment by joeyparrish** [comment]
Combined with PR #26, this completes phase 4, where we will have something like a single-sign-on system in the lab.

File: shaka-lab-gateway-client/README.md [hunk 1]
```diff
@@ -7,54 +7,11 @@ Lab Gateway to log in.
 [`shaka-lab-gateway`](https://github.com/shaka-project/shaka-lab/tree/main/shaka-lab-gateway#readme)
 package.  Devices with the client package will connect to that gateway.
 
+This package is available for:
```
Comment (theodab) [inline shaka-lab-gateway-client/README.md:10]: Do we still need the note about the gateway package in this file, if we also reproduce it in each platform-specific doc?

File: shaka-lab-gateway-client/macos/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+This is the macOS package.
+
+**NOTE**: Exactly one Linux device on your lab network must install the
+[`shaka-lab-gateway`](https://github.com/shaka-project/shaka-lab/tree/main/shaka-lab-gateway#readme)
+package.  Devices with the client package will connect to that gateway.
+
+## Pre-requisites
+
+ - Get Homebrew: [https://brew.sh/#install](https://brew.sh/#install)
+ - You must grant full disk access to Terminal before proceeding.  Without this
```
Comment (theodab) [inline shaka-lab-gateway-client/macos/README.md]: There's also the "brew doctor" command that gives you a list of specific folders you should change the ownership of (among other things).
Or would that not help for our case?

**Review by theodab — CHANGES_REQUESTED** [review]

**Review by alekzz — DISMISSED** [review]

File: shaka-lab-gateway-client/macos/README.md (hunk 2 above)
Comment (joeyparrish) [reply shaka-lab-gateway-client/macos/README.md]: No, I don't think that helps.  My understanding is that by default, you can't make changes to arbitrary files from Terminal until you do this, and this permission is needed for Homebrew or one of the scripts.

I can't recall the details, and it's always possible I was wrong.  I just got another Mac for the lab, though, so I'll leave this step in the docs for now, but I'll try without this step on a fresh device and see what breaks.  If nothing breaks, I'll remove this from the docs.  If something breaks, I'll add more details to the docs.

File: shaka-lab-gateway-client/README.md (hunk 1 above)
Comment (joeyparrish) [reply shaka-lab-gateway-client/README.md:10]: You can't use the gateway-client package without the gateway package running on Linux somewhere on the network.  I think if you're looking at the detailed doc for a specific platform, the note helps.  (Though I'm open to debate and changes if you disagree.)

File: shaka-lab-gateway-client/macos/README.md (hunk 2 above)
Comment (joeyparrish) [reply shaka-lab-gateway-client/macos/README.md]: I double-checked on a fresh install, and I found that gateway-client works fine without this.  In fact, it is only recommended-settings that requires Terminal to have full disk access, to change OS settings from the CLI.

So I'm removing this from the gateway-client doc, but leaving it in recommended-settings.

**Review by theodab — APPROVED** [review]

**Pull request merged**